homeassistant/components/xiaomi_aqara.py

```diff
@@ -162,12 +181,37 @@ def stop_ringtone_service(call):
         else:
             _LOGGER.error('Unknown gateway sid: %s was specified.', gw_sid)
 
-    hass.services.async_register(DOMAIN, 'play_ringtone',
-                                 play_ringtone_service,
-                                 description=None, schema=None)
-    hass.services.async_register(DOMAIN, 'stop_ringtone',
-                                 stop_ringtone_service,
-                                 description=None, schema=None)
+    def add_device_service(call):
+        """Service to add a new sub-device within the next 30 seconds."""
+        gw_sid = call.data.get(ATTR_GW_MAC).replace(":", "").lower()
+        for (_, gateway) in hass.data[PY_XIAOMI_GATEWAY].gateways.items():
+            if gateway.sid == gw_sid:
+                join_permission = {'join_permission': 'yes'}
+                gateway.write_to_hub(gateway.sid, **join_permission)
+                break
+        else:
+            _LOGGER.error('Unknown gateway sid: %s was specified.', gw_sid)
```

Comment:
Should we add a persistent notification telling the user to press a button on the device with in 30s ?

Example: https://github.com/home-assistant/home-assistant/blob/3430c1c8bc2b541b29024b633855ebe4cf4ca651/homeassistant/components/switch/broadlink.py#L106

Reply:
Good point! I will extend the PR.